app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala

```diff
@@ -44,12 +44,26 @@ class RegisterController @Inject()(val messagesApi: MessagesApi,
 
   import RegisterController.NSIUserInfoOps
 
-  def getConfirmDetailsPage: Action[AnyContent] = authorisedForHtsWithInfo {
+  def getConfirmDetailsPage(emailVerificationParams: Option[String]): Action[AnyContent] = authorisedForHtsWithInfo {
     implicit request ⇒
       implicit htsContext ⇒
         checkIfAlreadyEnrolled { _ ⇒
-          checkIfDoneEligibilityChecks { case (nsiUserInfo, _) ⇒
-            Ok(views.html.register.confirm_details(nsiUserInfo))
+          checkIfDoneEligibilityChecks { case (nsiUserInfo, _) ⇒ {
+            emailVerificationParams match {
+              case None ⇒ Ok(views.html.register.confirm_details(nsiUserInfo))
+              case Some(p) ⇒ {
+                val optionalParams = EmailVerificationParams.decode(p)
+                optionalParams.fold(BadRequest(views.html.register.email_verify_error("hts.email-verification.email-verify-error.bad-continue-url.content"))){ params ⇒
```

Comment:
We have a single page with a variety of error messages displayable. Plugging in the message key to denote the actual message is common in other repos I believe, and has the advantage of minimising the amount of code in the views where the tooling is less sophisticated.